### PR TITLE
ER-1446 Added check on vacancy owner in authorization

### PR DESCRIPTION
**Pull request by dev4valtech**
I have added check on ownership in the authorization, this should prevent provider from accessing vacancies that have been transferred. 

A related edge case is where a provider submits the vacancy and at the same time the employer revokes the permissions, the vacancy manages to pass through validation and gets submitted. 

To stop this, I have changed command to include the requesting owner and comparing that in the handler with what is on the vacancy. The command will fail in case the owner has changed.

### CODE REVIEW

File: src/Shared/Recruit.Vacancies.Client/Application/Exceptions/ExceptionMessages.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@ public static class ExceptionMessages
     {
         public const string VacancyUnauthorisedAccess = "The employer account '{0}' cannot access employer account '{1}' for vacancy '{2} ({3})'.";
         public const string VacancyUnauthorisedAccessForProvider = "The provider account '{0}' cannot access provider account '{1}' for vacancy '{2} ({3})'.";
+        public const string UserIsNotTheOwner = "The {0} user is not the owner of the vacancy";
```
[the remark this answers is not in the document]
Comment (dev4valtech) [reply src/Shared/Recruit.Vacancies.Client/Application/Exceptions/ExceptionMessages.cs]: The {0} is going to replace by "provider" or "employer" so the message is generic.

File: src/Provider/UnitTests/Provider.Web/Orchestrators/CloneVacancyOrchestratorTest/GetCloneableAuthorisedVacancyAsyncTests.cs [hunk 2]
```diff
@@ -18,8 +18,7 @@ public async Task WhenRouteHasInvalidUkprn_ShouldThrowAuthorizationException()
 
         [Theory]
         [InlineData(VacancyStatus.Referred)]
-        [InlineData(VacancyStatus.Draft)]
-        public async Task WhenVacancyInInvalidState_ShouldThrowInvalidStateException(VacancyStatus status)
+        [InlineData(VacancyStatus.Draft)]         public async Task WhenVacancyInInvalidState_ShouldThrowInvalidStateException(VacancyStatus status)
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/Provider/UnitTests/Provider.Web/Orchestrators/CloneVacancyOrchestratorTest/GetCloneableAuthorisedVacancyAsyncTests.cs]: Have you made any changes on this file? Not able to see them!

[the remark this answers is not in the document]
Comment (shomavg) [reply src/Provider/UnitTests/Provider.Web/Orchestrators/CloneVacancyOrchestratorTest/GetCloneableAuthorisedVacancyAsyncTests.cs]: still on the same line

File: src/Shared/Recruit.Vacancies.Client/Application/Exceptions/ExceptionMessages.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (shomavg) [reply src/Shared/Recruit.Vacancies.Client/Application/Exceptions/ExceptionMessages.cs]: 👍

File: src/Provider/UnitTests/Provider.Web/Orchestrators/CloneVacancyOrchestratorTest/GetCloneableAuthorisedVacancyAsyncTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (shomavg) [reply src/Provider/UnitTests/Provider.Web/Orchestrators/CloneVacancyOrchestratorTest/GetCloneableAuthorisedVacancyAsyncTests.cs]: still on the same line!

[the remark this answers is not in the document]
Comment (dev4valtech) [reply src/Provider/UnitTests/Provider.Web/Orchestrators/CloneVacancyOrchestratorTest/GetCloneableAuthorisedVacancyAsyncTests.cs]: Strange shows correctly on mine

[the remark this answers is not in the document]
Comment (dev4valtech) [reply src/Provider/UnitTests/Provider.Web/Orchestrators/CloneVacancyOrchestratorTest/GetCloneableAuthorisedVacancyAsyncTests.cs]: This is stale .. the file doesn't even appear on the changes.

**Pull request merged**